ui/app/src/components/DateTimePicker.tsx

```diff
@@ -100,97 +120,117 @@ interface timezoneType {
+      dateMoment = moment();
+  }
+
+  return dateMoment;
+};
+
+const DateTimePicker = withStyles(dateTimePickerStyles)((props: DateTimePickerProps) => {
+  const {
+    id,
+    classes,
+    onChange,
+    date = moment(),
+    disabled = false,
+    controls = ['date', 'time', 'timezone'],
+    datePickerProps = {},
+    timePickerProps = {},
+    timeZonePickerProps = {}
+  } = props;
+
+  // Set defaults on wrapped controls props
+  if (nou(timeZonePickerProps.timezone)) {
+    timeZonePickerProps['timezone'] = moment.tz.guess();
+  }
+  if (nou(datePickerProps.dateFormat)) {
+    datePickerProps['dateFormat'] = 'YYYY-MM-DD';
+  }
+  if (nou(datePickerProps.disablePast)) {
+    datePickerProps['disablePast'] = false;
+  }
+  if (nou(timePickerProps.timeFormat)) {
+    timePickerProps['timeFormat'] = 'HH:MM:ss';
   }
 
-  const timezoneObj = timezones.find( tz => (tz.timezoneName === timezone) );
-  const [ selectedDateTime, setSelectedDateTime ] = useState(initialDateMoment);
-  const [ selectedTimezone, setSelectedTimezone ] = useState(timezoneObj);
+  let dateMoment;
+  let timezoneObj = timezones.find(tz => (tz.timezoneName === unescape(timeZonePickerProps.timezone)));
+  dateMoment = getDateMoment(date, timezoneObj);
+  const { formatMessage } = useIntl();
 
-  const handleDateChange = (name: string) => (date: Date | null) => {
-    let updatedDateTime = selectedDateTime;
+  const handleDateChange = (name: string) => (newDate: Date | null) => {
+    let updatedDateTime = dateMoment;
 
     switch (name) {
       case 'scheduledDate':
-        updatedDateTime.date(date.getDate());
-        updatedDateTime.month(date.getMonth());
-        updatedDateTime.year(date.getFullYear());
+        updatedDateTime.date(newDate.getDate());
+        updatedDateTime.month(newDate.getMonth());
+        updatedDateTime.year(newDate.getFullYear());
+        datePickerProps?.onDateChange?.(updatedDateTime.format(datePickerProps.dateFormat));
         break;
       case 'scheduledTime':
-        updatedDateTime.hours(date.getHours());
-        updatedDateTime.minutes(date.getMinutes());
+        updatedDateTime.hours(newDate.getHours());
+        updatedDateTime.minutes(newDate.getMinutes());
+        timePickerProps?.onTimeChange?.(updatedDateTime.format(timePickerProps.timeFormat));
         break;
     }
 
-    setSelectedDateTime(updatedDateTime);
     onChange?.(updatedDateTime);
-
-    onChangeDate?.(updatedDateTime.format(dateFormat));
-    onChangeTime?.(updatedDateTime.format(timeFormat));
   };
 
   const handleTimezoneChange = () => (event: React.ChangeEvent<{}>, timezoneObj: any) => {
     const timezone = timezoneObj.timezoneName,
-          updatedDateTime = moment.tz(selectedDateTime.format(), 'YYYY-MM-DD HH:mm A', timezone);
-    setSelectedDateTime(updatedDateTime);
-    setSelectedTimezone(timezoneObj);
+      updatedDateTime = moment.tz(dateMoment.format(), `${datePickerProps.dateFormat} ${timePickerProps.timeFormat} A`, timezone);
```

Comment:
Is the trailing `A` at the end of the format string a typo?

Reply:
yeah seems like typo, removing